ansible_collections/ds389/ansible_ds/plugins/module_utils/ds389_entities.py

```diff
@@ -1034,15 +1111,94 @@ def _getChangelog(self):
+            if not hasattr(agmt, 'suffix'):
+                get_log().debug(f"ds389_agmts should have a 'suffix' option: {agmt}")
+                raise AttributeError("ds389_agmts should have a 'suffix' option")
+            if not hasattr(agmt, 'fulltargetname'):
+                get_log().debug(f"ds389_agmts should have a 'fulltargetname' option: {agmt}")
+                raise AttributeError("ds389_agmts should have a 'fulltargetname' option")
+            instname = agmt.fulltargetname.split('.')[-2]
+            if agmt.suffix == self.suffix and instname.lower() != self.parent().name.lower():
+                # Add agmt if suffix is ok and if it does not target itself
+                self.update_single_ds389agmt(agmt, facts, inst, onlycheck)
+
+    def update_replman(self, facts, inst, onlycheck):
+        """Creates of update Replication manager entry."""
+        del facts
+        cfginst = self.parent()
+        # Get dn
+        try:
+            adn = self.replicabinddn
+        except AttributeError:
+            errmsg = f"Option 'ReplicaBindDn' is missing in {cfginst.name}.{self.name} on host {ROOT_ENTITY}"
+            get_log().debug(f'update_replman: {errmsg}')
+            return
+        # Check if we can use that dn
+        match = re.fullmatch('cn=([^,]*),(([^,]*,)*cn=config)', adn)
+        if not match:
+            get_log().debug(f'update_replman: Unable to handle dn {adn}')
+            return
+        # extract cn and basedn
+        acn = match.group(1)
+        basedn = match.group(2)
+        # Get the password
+        try:
+            apw = self.replicacredentials
+        except AttributeError:
+            errmsg = f"Option 'ReplicaCredentials' is missing in {cfginst.name}.{self.name} on host {ROOT_ENTITY}"
+            get_log().debug(f'update_replman: {errmsg}')
+            get_log().error(f'{errmsg}')
+            raise AttributeError(errmsg) from None
+        account = ServiceAccount(inst, adn)
+        if account.exists():
+            if _is_password_ignored(cfginst, None, dn=adn, pw=apw):
+                # Password has not changed ==> nothing to do.
+                get_log().debug('update_replman: password is not changed.')
+                return
+            # Password has changed ==> update it.
+            self.add_change(f'Updating password on entry {adn}')
+            if not onlycheck:
+                get_log().debug(f'Updating password on entry {adn}')
+                account.reset_password(apw)
+            return
+        # Entry does not exist ==> Create it.
+        self.add_change(f'Adding entry {adn}')
+        if not onlycheck:
+            get_log().debug(f'create entry {adn}')
+            properties = {
+                'cn': acn,
+                'objectclass': [ 'netscapeServer', 'nsAccount', 'top' ],
+                'userpassword': apw,
+            }
+            account.create(rdn=f'cn={acn}', properties=properties, basedn=basedn)
```

Comment:
Minor thing, I think it makes sense to check if the entry is actually created: `account.exists():` and report a simple error if not.

Reply:
Not really sure about what you means exactly. But if you think we should not creates the entry, I disagree:
  if we do that there will be a systematic failure when creating an instance with a replication manager under cn=config, that said I think that we can do better that what is actually coded:
   Systematically test if the account exists and if it does not exists creates it if it is in cn=config and reports error otherwise

Reply:
No, no, what I meant was just a simple thought of rechecking that the server has the entry after the operation. Something like that:

```
        account.create(rdn=f'cn={acn}', properties=properties, basedn=basedn)
        if not account.exists():
            # Report an error and raise an exception
```

Another thing we can do is to use `ensure_state` method instead of `create`. 
https://github.com/389ds/389-ds-base/blob/main/src/lib389/lib389/_mapped_object.py#L1039

But I don't have a hard opinion on that. Up to you. :)

Reply:
I see, I just added an assert  account.exists() ( Frankly I am not sure it is very useful as we will get an exception if the ldap add operation fails (did not tries with account but during my tests I have seen it with be.create replica.create and agmt.create which are also based on _mapped_entries ...